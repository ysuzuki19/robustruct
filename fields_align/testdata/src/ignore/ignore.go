package ignore

type SampleStruct struct {
	Field1 int
	Field2 string
	Field3 bool
}

func DefinedUnaligned() SampleStruct {
	return SampleStruct{ // want "all fields of the struct must be sorted by defined order"
		Field2: "hello",
		Field1: 1,
		Field3: true,
	}
}

func Ignored() SampleStruct {
	// ignore:fields_align
	return SampleStruct{
		Field2: "hello",
		Field1: 1,
		Field3: true,
	}
}

func IgnoredEndOfStruct() SampleStruct {
	return SampleStruct{
		Field2: "hello",
		Field1: 1,
		Field3: true,
	} // ignore:fields_align
}
