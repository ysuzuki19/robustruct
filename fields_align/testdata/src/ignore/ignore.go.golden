package ignore

type SampleStruct struct {
	Field1 int
	Field2 string
	Field3 bool
}

func DefinedUnaligned() SampleStruct {
	return SampleStruct{
		Field1: 1,
		Field2: "hello",
		Field3: true,
	}
}

func Ignored() SampleStruct {
	// ignore:fields_align
	return SampleStruct{
		Field2: "hello",
		Field1: 1,
		Field3: true,
	}
}

func IgnoredEndOfStruct() SampleStruct {
	return SampleStruct{
		Field2: "hello",
		Field1: 1,
		Field3: true,
	} // ignore:fields_align
}
